plugins/pcap/main.cpp

```diff
@@ -446,15 +446,19 @@ class reader : public format::single_layout_reader {
       ts += microseconds(header->ts.tv_usec);
 #endif
       // Assemble packet.
-      const auto* layer3_ptr = reinterpret_cast<const char*>(layer3.data());
-      auto packet = std::string_view{std::launder(layer3_ptr), layer3.size()};
+      auto make_string_view = [](std::span<const std::byte> bytes) {
+        auto data = reinterpret_cast<const char*>(bytes.data());
+        auto size = bytes.size();
+        return std::string_view{data, size};
+      };
+      auto payload = make_string_view(raw_frame);
```

Comment:
It may be better to add an overload or generalize the `add` function for the builder to work with more contiguous types of characters (e.g. `std::span`) rather than requriing a `std::string_view` only.

Reply:
I thought about making that a generic utility, like we have for `as_bytes`. For example, `as_string_view` should work for any byte container. This way, we don't have to force the call site to be templated and pollute the headers.

Would you be on board with `as_string_view`?

Reply:
That would work, but I see an inconvenience that the `add` function only works with `std::string_view`. If it worked with spans for example, we wouldn't need to make a `std::string_view` at all.

Reply:
I think we can add an overload for byte spans as well, actually.

Reply:
The builder functions take a `data_view`, which is variant over concrete views. Unfortunately `std::string_view` is just an alias instead of a strong typedef. Otherwise we could construct a string view from a byte span just fine. We want to convert all elements of `data` and all views to strong typedefs in the medium term. Until then, I'd say we keep the verbosity at the call site.

Reply:
@JoeLoser (and @tenzir/backend) what do you think of an `as<T>` facility? With concepts, I think we can constrain it properly. In my eyes, it would make sense to have it as general-purpose, inexpensive (zero-cost, to be precise) conversion of "congruent" types, where accessing the same data is more convenient through a different API.